### ISSUE TITLE
Add pg_table_is_visible scalar, for Django and psql compatibility

### ISSUE THREAD

**Issue by robd003** [crate/crate#7988]
Crate almost works out of the box with Django 2.x when using the native Postgres backend except for a single case when Django tries to list all the tables.

It runs this query:

```sql
SELECT
    c.relname,
    c.relkind
FROM
    pg_catalog.pg_class c
    LEFT JOIN pg_catalog.pg_namespace n ON n.oid = c.relnamespace
WHERE
    c.relkind IN ('r', 'v')
    AND n.nspname NOT IN ('pg_catalog', 'pg_toast')
    AND pg_catalog.pg_table_is_visible(c.oid)

```

which fails on the last line due to the pg_catalog.pg_table_is_visible() function not existing. Crate will throw the following exception:

```
SQLActionException[UnsupportedFeatureException: unknown function: pg_catalog.pg_table_is_visible(integer)]
```

Could we make a "fake" function on the pg_catalog table that when called returns true? Another option would be to check if the pg_catalog.pg_class oid exists and then return true.

---

**Maintainer edits**

Also used by psql:

```
crate=> \d
ERROR:  Unknown function: pg_catalog.pg_table_is_visible(c.oid)
```

**Comment by mfussenegger** [comment]
That's something we want to add eventually as we've seen it being used by other postgres tools as well.

Would you be interested in contributing a django test case to [crate-qa](https://github.com/crate/crate-qa), so that we've a test case that triggers this query and can verify that it works correctly once we implement it?

**Comment by robd003** [comment]
@mfussenegger Sure, let me look at crate-qa and I'll try to get a minimal Django test working

**Comment by mfussenegger** [comment]
@robd003 any update on the test-case for crate-qa?

**Comment by robd003** [comment]
@mfussenegger I'll get this out on the weekend, will submit pull request to crate-qa

**Comment by amotl** [comment]
`pg_table_is_visible` is also needed by pandas when using its `to_sql` primitive to talk to CrateDB's PostgreSQL server interface.

- https://github.com/pyveci/sqlalchemy-postgresql-relaxed/issues/33

**Comment by hlcianfagna** [comment]
@robd003 for the purpose of the `SELECT` query you shared it may be enough to mock the function like below, could you give this a try?
```sql
CREATE FUNCTION pg_catalog.pg_table_is_visible(oid int) 
RETURNS BOOLEAN 
LANGUAGE JAVASCRIPT 
AS 'function pg_table_is_visible(oid){return true;}';
```

**Comment by robd003** [comment]
@hlcianfagna sure, giving this a try on Django 5.0

**Comment by jeeminso** [comment]
Thank you @robd003 , `pg_table_is_visible` function will be available in `5.8` release.
